ci-operator/config/openshift/enhancements/openshift-enhancements-master.yaml

```diff
@@ -0,0 +1,42 @@
+base_images:
+  os:
+    name: fedora
+    namespace: openshift
+    tag: "29"
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.14
+raw_steps:
+- pipeline_image_cache_step:
+    commands: |
+      cat <<DOCKERFILE > /opt/Dockerfile
+      FROM os
+      RUN dnf -y module enable nodejs:12 && dnf -y install nodejs
+      RUN npm install -g markdownlint markdownlint-cli2
+      DOCKERFILE
+    from: os
+    to: markdownlint
+- project_directory_image_build_step:
```

Comment:
The `project_directory_image_build_step:` is not needed. Generally, `images` is what is supposed to be used to build images: https://steps.ci.openshift.org/ci-operator-reference#line72 and `raw_steps` is not really supposed to be set by users. I would also supposed that is how things go wrong here.

Reply:
I don't really know what this means.

I don't want to publish any images out of this repo. I only need the container where the test job is running to have some extra tools (not written in go) in it. The image can be thrown away when the job is done.

I do have a Dockerfile in the openshift/enhancements repository, but the reference guide at https://steps.ci.openshift.org/ci-operator-reference isn't clear about how I can use that to achieve my goal. Do I want to make a new `build_root`? I tried

```
build_root:
  project_image:
    dockerfile_path: hack/Dockerfile.markdownlint
```

but that didn't seem to work.

I also tried

```
images:
  - dockerfile_path: hack/Dockerfile.markdownlint
    from: os
    to: markdownlint
```

but that didn't work either.

You suggested running the command to install the package I need within the job's command string, but since I need to install an RPM as root I don't think that approach will work.

I'm sure the problem is that I don't have a good mental model of how these input files translate into the steps the CI system actually runs. I've tried reading the documentation at https://docs.ci.openshift.org but I don't have enough context to understand them.

Maybe I'm trying to do something the system isn't designed to support?